scpdt/plugin.py

```diff
@@ -54,11 +56,50 @@ def pytest_ignore_collect(collection_path, config):
         if "tests" in path_str or "test_" in path_str:
             return True
 
-    for entry in config.dt_config.pytest_extra_skips:
+    for entry in config.dt_config.pytest_extra_ignore:
         if entry in str(collection_path):
             return True
 
 
+def is_private(item):
+    """Decide if an DocTestItem `item` is private.
+
+    Private items are ignored in pytest_collect_modifyitem`.
+    """
+    # Here we look at the name of a test module/object. A seemingly less
+    # hacky alternative is to populate a set of seen `item.dtest` attributes
+    # (which are actual DocTest objects). The issue with that is it's tricky
+    # for explicit skips/ignores. Do we skip linalg.det or linalg._basic.det?
+    # (collection order is not guaranteed)
+    parent_full_name = item.parent.module.__name__
+    is_private = "._" in parent_full_name
+    return is_private
+
+
+def _maybe_add_markers(item, config):
+    """Add xfail/skip markers to `item` if DTConfig says so.
+
+    Modifies the item in-place.
+    """
+    dt_config = config.dt_config
+
+    extra_skip = dt_config.pytest_extra_skip
+    skip_it = item.name in extra_skip
+    if item.name in extra_skip:
```

Comment:
`skip_it` is a variable created but not used. Can be addressed in a follow-up PR though.